atoma-proxy-service/src/components/grafana.rs

```diff
@@ -0,0 +1,207 @@
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use thiserror::Error;
+use tracing::instrument;
+
+/// A client for interacting with Grafana
+///
+/// This struct provides methods for querying data from Grafana dashboards.
+#[derive(Clone)]
+pub struct Grafana {
+    /// The URL of the Grafana instance
+    url: String,
+    /// The API token to use for authentication
+    api_token: String,
+    /// The tag to use to filter dashboards
+    dashboard_tag: String,
+}
+
+#[derive(Deserialize)]
+struct DashboardList {
+    uid: String,
+}
+
+#[derive(Deserialize)]
+struct Time {
+    from: String,
+    to: String,
+}
+
+#[derive(Deserialize)]
+struct Panel {
+    targets: Value,
+    title: String,
+}
+
+#[derive(Deserialize)]
+struct InnerDashboard {
+    panels: Vec<Panel>,
+    time: Time,
+    title: String,
+}
+
+#[derive(Deserialize)]
+pub struct Dashboard {
+    dashboard: InnerDashboard,
+}
+
+impl Dashboard {
+    pub fn title(&self) -> String {
+        self.dashboard.title.clone()
+    }
+}
+
+#[derive(Debug, Serialize)]
+struct Query {
+    queries: Value,
+    from: String,
+    to: String,
+}
+
+/// Convert a `Dashboard` into a vector of queries
+impl From<Dashboard> for Vec<(String, Query)> {
+    fn from(dashboard: Dashboard) -> Self {
+        let from = dashboard.dashboard.time.from;
+        let to = dashboard.dashboard.time.to;
+        dashboard
+            .dashboard
+            .panels
+            .into_iter()
+            .map(|panel| {
+                (
+                    panel.title,
+                    Query {
+                        queries: panel.targets,
+                        from: from.clone(),
+                        to: to.clone(),
+                    },
+                )
+            })
+            .collect()
+    }
+}
+
+impl Grafana {
```

Comment:
nitty: As a matter of taste, I usually prefer implementation bodies to be close to the struct declaration